src/Common/ThreadStatus.cpp

```diff
@@ -199,13 +199,12 @@ ThreadStatus::~ThreadStatus()
     if (deleter)
         deleter();
 
+    assert(!check_current_thread_on_destruction || current_thread == this);
```

Comment:
Maybe we can also add `LOG_ERROR` if `!check_current_thread_on_destruction || current_thread == this` to track this in release build. I am trying to debug https://github.com/ClickHouse/ClickHouse/issues/51780 but such failures occures only in release builds (maybe with sanitizers)

Reply:
throw `LOGICAL_ERROR` maybe?

Reply:
From destructor? Then we will have possible aborts in release build

Reply:
yeah, but I saw `adjustWithUntrackedMemory` so thought maybe there is no difference
will add `LOG_ERROR`